deps/NNNN-observability.md

```diff
@@ -0,0 +1,168 @@
+# Common Observability Framework
+
+**Status:** Draft
+
+**Authors:** nnshah1, keivenchang, whoisj
+
+**Category:** Architecture
+
+**Replaces:** N/A
+
+**Replaced By:** N/A
+
+**Sponsor:** keivenchang
+
+**Required Reviewers:** TBD  
+
+**Review Date:** [Date for review]
+
+**Pull Request:** [Link to Pull Request of the Proposal itself]  
+
+**Implementation PR / Tracking Issue:** TBD  
+
+# Summary
+
+This document outlines and defines the Common Observability Framework, detailing its goals and test cases. The DOF allows Dynamo programmers to add and expose observability throughout the system in a common, consistent, and performant manner, avoiding the use of disparate raw libraries that could compromise safety, performance, consistency, and maintainability.
+
+# Motivation
+
+## Goals
+
+The Common Observability Framework in Dynamo is designed to improve system visibility and support greater fault tolerance. Relying on various different libraries often results in compatibility issues as well as performance and safety issues, which can impact service reliability. By providing a common framework, we simplify implementation, reduce development time, and promote best practices in observability.
+
+* Provide a common metrics endpoint for each component/process
+* Provide common APIs for programmers to add metrics and tracing
+* Provide a way to create a common data formats (e.g., structs and metric types)
+* Provide flexibility – allow different implementations using the same APIs
+* Improve safety: use tested and well-known APIs instead of disparate libraries
+
+## Non-Goals
+
+* Legacy support (e.g., Triton)
+* Malicious actor resistance (DDOS protection) on /metrics endpoints
+* For now, we are not focused on logging.
```

Comment:
lets add a section on logging as well - I think we should have three apis: metrics, tracing, logging -